pkg/liqoctl/info/utils.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package info
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+// collectData collect the data retrieved by the checkers in a map.
+func (o *Options) collectData(checkers []Checker) map[string]interface{} {
+	data := map[string]interface{}{}
+
+	for i := range checkers {
+		data[checkers[i].GetID()] = checkers[i].GetData()
+	}
+
+	return data
+}
+
+// installationCheck checks if Liqo is installed in the cluster.
+func (o *Options) installationCheck(ctx context.Context) error {
+	if _, err := o.KubeClient.CoreV1().Namespaces().Get(ctx, o.LiqoNamespace, metav1.GetOptions{}); client.IgnoreNotFound(err) != nil {
+		o.Printer.Error.Printf("Unable to check if Liqo is installed: %v", err)
+		return err
+	} else if err != nil {
+		o.Printer.Error.Println("Liqo is not installed in the current cluster! \n\n" +
+			"You can install liqo via the 'liqoctl install' command.\n" +
+			"Check 'liqoctl install --help' for further information.")
+		return err
+	}
+
+	return nil
```

Comment:
I know it works like this, but in general we handle this with a switch to be more clear and comprensible what is the if-else block that handles a NotFound error.